hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -488,8 +495,9 @@ private void initTableIfNotExists(ObjectPath tablePath, CatalogTable catalogTabl
     }
   }
 
-  private String inferTablePath(ObjectPath tablePath, CatalogBaseTable table) {
-    String location = table.getOptions().getOrDefault(PATH.key(), "");
+  @VisibleForTesting
+  public String inferTablePath(ObjectPath tablePath, CatalogBaseTable table) {
```

Comment:
Revert this change, the `table` should never be null and we should respect the path of the table options.